### PR TITLE
feat: improve the preview script and related documentation

### PR DESCRIPTION
**Pull request by tbouffard**
Improvements
  - enforce 'redirect' and 'no fetch sources' config except if explicitly
  requested
  - allow to build all components/branches using a new `use-all-repositories`
  option
  - allow to build a set of components and branches with new options
  (`use-multi-repositories` and `component-with-branches`)
  - add a dedicated bash preview script for local developments. Everything can
  now be done with the script so specific npm scripts have been removed from the
  package.json file
  - new bash script option to skip the doc generation and only generate the
  Antora preview playbook
  - improve the site title management logic
  - allow to use boolean options without having to pass 'true' as argument

Fixes
  - allow the bonita '7.10' branch build
  - make the 'fetch sources' option work

Readme updates
  - reorganize to put end-user documentation first
  - put local developments information first as this is what developers mainly
  use
  - provide more details about the `build-preview` bash script usage (for
  instance http server and asciinema)

### CODE REVIEW

**Comment by tbouffard** [comment]
The new options to build a set of components and branches has been tested on https://github.com/bonitasoft/bonita-continuous-delivery-doc/pull/164/ to check cross component xref

```bash
./build-preview-dev.bash --fetch-sources false --local-sources \
  --use-multi-repositories \
  --component-with-branches bcd:fix/DS-52_asciidoc_attribute_substitution_and_broken_links \
  --component-with-branches bonita:7.11,2021.1
```

File: README.adoc [hunk 1]
```diff
@@ -17,97 +17,124 @@ endif::[]
...
+* and much more
 
 
 == Build
 
 === Requirements
 
 
-{url-node}[Node 12.x] (may work with newer versions, no guarantee). For more details, see the  https://docs.antora.org/antora/2.3/install-and-run-quickstart/[Antora installation guide].
+{url-node}[Node 12.x] (may work with newer versions, no guarantee).For more details, see the  https://docs.antora.org/antora/2.3/install-and-run-quickstart/[Antora installation guide].
```
Comment (tbouffard) [inline README.adoc:39]:
```suggestion
{url-node}[Node 12.x] (may work with newer versions, no guarantee). For more details, see the  https://docs.antora.org/antora/2.3/install-and-run-quickstart/[Antora installation guide].
```

File: build-preview.bash [hunk 2]
```diff
@@ -7,56 +7,72 @@ set -euo pipefail
...
-  echo "        --help                                  Display this help"
+  echo "Usage: ${launch_command} <options>"
+  echo "Build the documentation site preview."
+  echo ""
+  echo "For 'boolean' options, you can only pass the options when you want to set it to 'true'."
+  echo "i.e '--force-production-navbar' is equivalent to '--force-production-navbar true'."
+  echo ""
+  echo "Options priority order for choosing the content of the preview when several options are provided:"
+  echo "  - use-all-repositories"
+  echo "  - single-branch-per-repo"
```
Comment (tbouffard) [inline build-preview.bash:18]:
```suggestion
  echo "  - single-branch-per-repo"
  echo "  - use-multi-repositorie"
```

File: README.adoc [hunk 3]
```diff
@@ -17,97 +17,124 @@ endif::[]
...
+The easiest way is to use a script that generates the documentation partially and/or with specific settings for both local
+developments, Pull Request previews or to produce documentation archives.
 
+This includes building the documentation using:
 
-==== Building for local developments
+* already checkout local documentation content repositories
+* single branch for all components
+* single branch of a single component
+* ....
```
Comment (tbouffard) [inline README.adoc:69]:
```suggestion
* a set of branches of various components. This is let validate https://docs.antora.org/antora/2.3/page/version-and-component-xrefs/[references between pages that belong to different docs components and/or versions]
* ....
```

File: README.adoc [hunk 4]
```diff
@@ -17,97 +17,124 @@ endif::[]
...
+==== Using local documentation content repositories
 
-This activates the https://docs.antora.org/antora/2.3/playbook/author-mode/[Antora author mode]. The local content of the
-repository is used instead of fetching the remote one.
+This activates the https://docs.antora.org/antora/2.3/playbook/author-mode/[Antora author mode]. +
+Antora uses the locally cloned repositories as they are on your local machine instead of cloning them from their remote location.
 
-It provides a fast feedback loop as you don't need to push nor commit content as the current work
+This provides a fast feedback loop as you don't need to push nor commit sources content as the current work or make developments
+easier when working offline.
```
Comment (tbouffard) [inline README.adoc:101]:
```suggestion
This provides a fast feedback loop as you don't need to push nor commit sources content. This also makes developments easier when working offline.
```

File: README.adoc [hunk 5]
```diff
@@ -33,81 +41,100 @@ Static site generation based on {url-antora}[Antora] using {url-asciidoctor}:[As
...
-====
-The following command https://docs.antora.org/antora/2.3/playbook/runtime-fetch/[fetch the documentation content repositories] each time they are run. +
-<<local-content>> or <<preview>> can prevent this to happen.
-====
+To know all available options, run
+[source,shell script]
+----
+./build-preview.bash --help
+----
+
```
Comment (alachambre) [inline README.adoc:77]: I would put around here an example of the command line (copy past ready) to build and preview the site for a given branch with local sources -> It's an interesting use case IMHO, it allows contributors to have a preview without opening a pull request, which is what you wanna do when you are working on a big contribution

Comment (tbouffard) [reply README.adoc:77]: yes good idea, I will also put the example to build cross component versions like in https://github.com/bonitasoft/bonita-documentation-site/pull/222#issuecomment-814609850

Comment (tbouffard) [reply README.adoc:77]: done in 4227066, feel free to update the example and explanation if it is uncleared. As I have developed the script and use it a lot, I am biased and don't know really what is needed to let a new comer understand how it works 😄

**Review by benjaminParisel — APPROVED** [review]

**Pull request merged**